activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -642,6 +644,14 @@ def first_nth_or_last(type, *args)
             set_inverse_instance record if record.is_a? ActiveRecord::Base
           end
         end
+
+        def associate_all_records(target)
+          if owner.new_record? && target.all? {|record| record.new_record? }
```

Comment:
I'd rather do the association when the `target` contains all new records as they will all be associated with the right instance of the `owner`.  

But at the same time I may be wrong here and there might not be a with only associating the new ones. 

Can you give me a reason to only associate the new one's?